### DISCUSSION TITLE
XuanFang internal connector

### DISCUSSION THREAD

**Discussion by fireball2k2000** [mathoudebine/turing-smart-screen-python#233] [General]
Hi everyone,

thanks for this great project. Does anyone know what type the "internal" connector on the XuanFang display is? I'd like to install the display internally on my machine without having the USB connector sticking out.

Best regards
Marcus

**Comment by mathoudebine** [comment]
Hi, it might be one of those 3 connectors but not sure, i do not have them at home to test
![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/38615348/c943aa41-1628-4e9d-bd81-e4a8890387ae)

**Comment by fireball2k2000** [comment]
Hm - yes... probably the MX-1.25. I'll give it a try, Thanks :)

I'll give an update as soon as the connectors arrive.

**Comment by fireball2k2000** [comment]
Well... MX-1.25 (GH) has a too small contact pitch. Next try... JST ZH...

**Comment by fireball2k2000** [comment]
JST ZH is working fine. Needs a bit fiddling to get it in, but it works :)

(USB Voltage and Data Pins are directly connected to the JST Connector)

**Comment by Kenny-Dave** [comment]
Is the connector on a Turzx 5" likely to also be JST ZH?

I want to connect it to a USB port, the connection through the normal USB connection is flaky if I move the unit at all. I don't know if this is even possible, or how I would wire it in. I can't see a prebuilt cable for what I'm looking for. 

[Image](https://i.imgur.com/vDiEzn2.png).

**Reply by mathoudebine** [reply]
I have the exact same problem with my Turing 5"! The USB-C connector cannot be used at all, I had to rely on the connector from your photo.
It is not the same connector as the XuanFang, so I don't think it is JST ZH.
From the picture above, I would say the Turing 5" connector is a MX-1.25
Fortunately for me, the USB cable with the connector was provided when I bought the product

**Reply by Kenny-Dave** [reply]
Oh nice one, I didn't think to check that. I have the cable in the box too. 

It goes to a motherboard 5 pin header, not a USB. But I think I've found a cable for that. USB 2 only though, that shouldn't make a difference I think? 

It's so frustrating that there isn't tech sheets for all this. Thank you for your help.

**Reply by mathoudebine** [reply]
USB 2 is perfect, smart screens do not support USB 3
You can find this type of cable from your favorite online retailer:
![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/38615348/d95dd8c5-f9d8-4a49-a7e8-338b4afc2fe2)
E.g. on Aliexpress:
https://aliexpress.com/item/1005005301963316.html

**Comment by Kenny-Dave** [comment]
Hmm, absolutely nothing happening on the cable when connected. I've got one dodgy port, and one that doesn't work at all. You've got one dodgy port. 

I am not sure the build quality is all that great on these units then.
